### PR TITLE
Adding view to user bookmark count

### PR DESCRIPTION
**Pull request by sambuddhabasu**

### CODE REVIEW

File: bookie/static/js/bookie/api.js [hunk 1]
```diff
@@ -694,6 +694,27 @@ YUI.add('bookie-api', function (Y) {
...
+     *
+     * @class Api.route.UserBmarkCount
+     * @extends Api.route
+     *
+     */
+    Y.bookie.Api.route.UserBmarkCount = Y.Base.create(
+        'bookie-api-route-user-bmark-count',
+        Y.bookie.Api.route,
+        [], {
+            initializer: function (cfg) {
```
Comment (mitechie) [inline bookie/static/js/bookie/api.js:708]: the open/close can just be on one line: {}

File: bookie/static/js/bookie/api.js [hunk 2]
```diff
@@ -694,6 +694,27 @@ YUI.add('bookie-api', function (Y) {
...
     );
 
+    /**
+     * Fetch user bookmark count over a period of time
+     *
+     * @class Api.route.UserBmarkCount
+     * @extends Api.route
+     *
+     */
+    Y.bookie.Api.route.UserBmarkCount = Y.Base.create(
```
Comment (mitechie) [inline bookie/static/js/bookie/api.js:704]: never mind, ignore me. This is the api layer.

File: bookie/static/js/bookie/view.js [hunk 3]
```diff
@@ -958,6 +958,121 @@ YUI.add('bookie-view', function (Y) {
...
+         * @method _fetch_dataset
+         * @private
+         *
+         */
+        _fetch_dataset: function(data) {
+            var that = this;
+            this.api = new Y.bookie.Api.route.UserBmarkCount(this.get('api_cfg'));
+            this.api.call({
+                success: function(data, request) {
+                    //Stores the data for the graph
```
Comment (mitechie) [inline bookie/static/js/bookie/view.js]: space after the // and end in a period

File: bookie/static/js/bookie/view.js [hunk 4]
```diff
@@ -958,6 +958,121 @@ YUI.add('bookie-view', function (Y) {
...
+                    var myDataValues = [];
+                    for(var i=0;i<data.count.length;i++) {
```
Comment (mitechie) [inline bookie/static/js/bookie/view.js]: space after the for and each semicolon

Comment (mitechie) [inline bookie/static/js/bookie/view.js]: hmm, javascript has a forEach we should be able to use. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach

File: bookie/static/js/bookie/view.js [hunk 5]
```diff
@@ -958,6 +958,121 @@ YUI.add('bookie-view', function (Y) {
...
+                    xAxis: "dateRange",
+                    yAxis: "bookmarks",
+                    xKey: "date",
+                    yKey: "bookmarks",
+                    xDisplayName: "Date",
+                    yDisplayName: "Bookmarks"
+                }
+            ];
+
+        },
```
Comment (mitechie) [inline bookie/static/js/bookie/view.js:1107]: blank line before initializer

File: bookie/templates/stats/userstats.mako [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+}
+</style>
+<div class="bmarks"></div>
+<%include file="../jstpl.mako"/>
+<p>Bookmark count stats for ${request.user.username}</p>
+<span id="user_stats"></span>
+<div id="mychart"></div>
+<%def name="add_js()">
+<%
+import json
```
Comment (mitechie) [inline bookie/templates/stats/userstats.mako]: I don't think you need this json bits

File: bookie/templates/stats/userstats.mako [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+    <script type="text/javascript">
+        // Create a new YUI instance and populate it with the required modules.
+        YUI().use('bookie-view', 'charts', function (Y) {
+            Y.on('domready', function() {
+                var api_cfg = {
+                    url: APP_URL + '/api/v1',
+                    username: '${request.user.username}',
+                    api_key: '${request.user.api_key}',
+                },
+                user_stats_view = new Y.bookie.UserBmarkCountView({
```
Comment (mitechie) [inline bookie/templates/stats/userstats.mako]: need a var here. 

```
var user_stats_view = 
```

File: bookie/templates/stats/userstats.mako [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+                    api_cfg: api_cfg
```
Comment (mitechie) [inline bookie/templates/stats/userstats.mako]: try adding another bit to the object here.

```
api_cfg: api_cfg,
container: Y.one('body')
```

File: bookie/static/js/bookie/view.js [hunk 9]
```diff
@@ -958,6 +958,185 @@ YUI.add('bookie-view', function (Y) {
...
+                    var user_graph = Y.one('#bmark_count_graph');
+                    user_graph.setContent();
+                    var graph_render = new Y.Chart({
+                        dataProvider: myDataValues,
+                        axes: that.myAxes,
+                        seriesCollection: that.seriesCollection,
+                        render: user_graph
+                    });
+		},
+                error: function (data, status_str, response, args) {
```
Comment (mitechie) [inline bookie/static/js/bookie/view.js:1025]: we should at least output a console.log() about the error to aid in debugging.

**Comment by mitechie** [comment]
I've landed this for now to get started and see some data. We should continue to add the error reporting and tests making sure that everything is covered.

**Comment by sambuddhabasu** [comment]
Sure, the error reporting and the tests will be added.

**Pull request closed without merging**